src/utilcode/util.cpp

```diff
@@ -1312,6 +1314,29 @@ DWORD_PTR GetCurrentProcessCpuMask()
 #endif
 }
 
+size_t GetOsPageSizeUncached()
+{
+    SYSTEM_INFO sysInfo;
+    ::GetSystemInfo(&sysInfo);
+    return sysInfo.dwAllocationGranularity ? sysInfo.dwAllocationGranularity : 0x1000;
+}
+
+size_t GetOsPageSize()
+{
+#ifdef FEATURE_PAL
+    // Initialization of function-local statics is guaranteed to occur only
+    // once even when called from multiple threads (from http://en.cppreference.com/w/cpp/thread/call_once)
+    //
+    // However there is a real potential performance cost.  This will need to
+    // be optimized if GetOsPageSize() shows up in profiles.
+    static const size_t pageSize(GetOsPageSizeUncached());
```

Comment:
@jkotas It was not obvious how to play the same initialization trick for util.cpp.  Therefore I looked at @swgillespie 's call_once suggestion which suggested function-local statics are preferred over call_once.  

Based on my understanding of the function-local static requirements the above code is guaranteed to be functionally correct.  But as you noted performance may be an issue.